src/server/system_services/system_store.js

```diff
@@ -364,10 +364,13 @@ class SystemStore extends EventEmitter {
     [util.inspect.custom]() { return 'SystemStore'; }
 
     initial_load() {
+        const { NOOBAA_ROOT_SECRET } = process.env;
+        if (NOOBAA_ROOT_SECRET) this.master_key_manager.load_root_key();
         db_client.instance().on('reconnect', () => this.load());
         P.delay(100)
-            .then(() => {
+            .then(async () => {
                 if (db_client.instance().is_connected()) {
+                    if (!NOOBAA_ROOT_SECRET) await this.master_key_manager.load_root_keys_from_mount();
```

Comment:
we should handle failures here,  retry, catch with proper error, maybe panic. we need to think about it.